### ISSUE TITLE
Upload script should use dotrun to build

### ISSUE THREAD

**Issue by anthonydillon** [canonical-web-and-design/maas-ui#4163]
**Describe the bug**
Upload script should use dotrun to build before uploading to ensure dotrun works with the project.

**Comment by huwshimi** [comment]
@anthonydillon I'm not sure if the upload script is the right place to do that - the uploaded file won't be run using dotrun. @petermakowski has a PR to run `dotrun` in an action: https://github.com/canonical-web-and-design/maas-ui/pull/4125.

**Comment by anthonydillon** [comment]
You are using yarn build to build the package and I am suggesting we should use the recommended dotrun method. We would just get the side effect of failing if it does in dotrun were not compatible anymore.

**Comment by huwshimi** [comment]
@anthonydillon I wonder if we'd be better off changing our build and test steps in the main action to use dotrun which means we'd catch these errors when a PR is created rather than when the PR is merged (which is when the upload action runs).

We run the same build step for PRs here: https://github.com/canonical-web-and-design/maas-ui/blob/main/.github/workflows/test.yml#L56

**Comment by anthonydillon** [comment]
Yeah that would be good and align with other projects maintained by the team.
